pypechain/test/events/types/EventsContract.py

```diff
@@ -176,34 +178,64 @@ class EventsEventAContractEvent(ContractEvent):
     def __init__(self, *argument_names: tuple[str]) -> None:
         super().__init__(*argument_names)
 
-    def get_logs(  # type: ignore
+    # We ignore types here for function redefinition
+    def get_typed_logs(  # type: ignore
         self: "EventsEventAContractEvent",
         argument_filters: dict[str, Any] | None = None,
         from_block: BlockIdentifier | None = None,
         to_block: BlockIdentifier | None = None,
         block_hash: HexBytes | None = None,
-    ) -> Iterable[EventData]:
-        return cast(
-            Iterable[EventData],
-            super().get_logs(
-                argument_filters=argument_filters, from_block=from_block, to_block=to_block, block_hash=block_hash
-            ),
+    ) -> Iterable[EventAEvent]:
+        """Extension of `get_logs` that return a typed dataclass of the event."""
+        abi_events = super().get_logs(
+            argument_filters=argument_filters, from_block=from_block, to_block=to_block, block_hash=block_hash
         )
+        # TODO there may be issues with this function if the user uses a middleware that changes event structure.
```

Comment:
can you make an issue for this? I think some sort of run-time check is the right move here, but we can deal with it later.

Reply:
https://github.com/delvtech/pypechain/issues/131